dev/Repeater/APITests/ViewManagerTests.cs

```diff
@@ -586,6 +586,43 @@ public void CanResetLayoutAfterUniqueIdReset()
             });
         }
 
+        [TestMethod]
+        public void ValidateDataContextDoesNotGetOverwritten()
+        {
+            ItemsRepeater repeater = null;
+            const string c_element1DataContext = "Element1_DataContext";
+
+            RunOnUIThread.Execute(() =>
+            {
+                List<Button> data = new List<Button>();
+
+                var element1 = new Button();
+                element1.Content = "Element1_Content";
+                element1.DataContext = c_element1DataContext;
+                data.Add(element1);
+
+                var elementFactory = new DataAsElementElementFactory();
+
+                repeater = new ItemsRepeater() {
+                    ItemsSource = data,
+                    ItemTemplate = elementFactory,
```

Comment:
> ItemTemplate = elementFactory, [](start = 19, length = 31)

If ItemsSource contains UIElements, we don't call the element factory anymore with these changes, so this will not be used.

Reply:
The element factory is always called, even if the source has UIElements.

---
In reply to: [355564736](https://github.com/microsoft/microsoft-ui-xaml/pull/1740#discussion_r355564736) [](ancestors = 355564736)

Reply:
Yep, if there is an itemTemplateFactory set, the current code always retrieves an element from the factory.